pages/instructorAssessmentInstances/instructorAssessmentInstances.sql

```diff
@@ -9,9 +9,25 @@ SELECT
     CASE
         WHEN ai.open AND ai.date_limit IS NOT NULL
             THEN greatest(0, floor(extract(epoch from (ai.date_limit - current_timestamp)) / 60))::text || ' min'
-        WHEN ai.open THEN 'Open'
+        WHEN ai.open THEN 'Unlimited'
```

Comment:
There are two different concepts here, close vs open and time limits. Re-opening an exam will set the time limit to unlimited (this is sorta mentioned in the popover for re-open as well, though I can make it clearer).

Reply:
An alternative is to have "Re-open" open the time limit popover and then update the time limit in the same operation as re-open. Then a closed assessment would show "Re-open", while an open assessment would should "Close" or "Change Time Limit".

Reply:
To make sure I'm understanding, at the moment we have both `open` and `date_limit` fields. The useful combinations of these are:
1. `open=true` and `date_limit=null`: Assessment is open without a timer (or with an unlimited timer, to say it a different way)
2. `open=true` and `date_limit > now()`: Timer is counting down
3. `open=true` and `date_limit < now()`: Timer has expired, the assessment will close shortly
4. `open=false`: Assessment is closed (timer is irrelevant)

I think we are talking about what to call case (1) above?

Until now we've been saying that case (1) is "Open" and case (2) is "Closed". I agree that, especially once #3622 is implemented, we'll have to change these terms.

Could we wait until #3622 before making the change "Open" -> "Unlimited"? When #3622 is implemented we'll also change the "Re-open"/"Close" buttons to better names, and we can make sure everything is consistent. Until we do that I'd prefer to avoid terminology/UI changes.

Reply:
I think making the distinction between Open and Unlimited is probably a good idea here, to make it clear that the exam is not only open, but also has no time limit. An alternative is to list this as "No time limit".

Reply:
My concern is how a user is supposed to know that "Unlimited" also means "Open".

I could live with "Open (no time limit)" as a temporary fix. What do you think? I agree that it's basically all very confusing until we get #3622 done.

Once we do something about #3622 then I guess we'll have this column show "No time limit", "5 min", or "Expired"? With actions like "Remove time limit", "Change time limit", "Expire time limit"? And no mention of "open"/"closed" at all?

Reply:
One option, in preparation for #3622, is to include the "Expired" option if the time expired but the exam is not closed yet (possibly because the student's browser didn't trigger the close yet, or the student closed the tab without closing the assessment). I'm also fine with renaming "Set to unlimited" to "Remove time limit", and add an "Expire time limit" option to the dropdown. Then remove the "Re-open" option (since everything that needs to be done in that button is already possible in the Change time limit option), so that all actions in the "Actions" dropdown are related to grading (for now Close & Grade and Regrade, but based on #3644 could also include an option to grade saved answers without closing or affecting the time limit).

This would also affect the "Re-open all" option, so for consistency I can add the same options in the "Change all time limits", and add a flag there to indicate if it should also be applied to closed and/or expired assessments.